test/src/test/java/org/corfudb/utils/LockStoreTest.java

```diff
@@ -0,0 +1,283 @@
+    private boolean renewed1 = false;
+    private boolean renewed2 = false;
+    private List<LockDataTypes.LockId> expiredLockIds1 = new ArrayList<>();
+    private List<LockDataTypes.LockId> expiredLockIds2 = new ArrayList<>();
+    private CorfuRuntime runtime1;
+    private CorfuRuntime runtime2;
+    private final LockDataTypes.LockId lockId = LockDataTypes.LockId.newBuilder()
+        .setLockGroup(LOCK_GROUP).setLockName(LOCK_NAME).build();
+
+    @Before
+    public void setUp() {
+        runtime1 = getDefaultRuntime().connect();
+        runtime2 = getNewRuntime(getDefaultNode()).connect();
+    }
+
+    @After
+    public void cleanUp() {
+        clearLockTable(runtime1);
+        acquired1 = false;
+        acquired2 = false;
+        renewed1 = false;
+        renewed2 = false;
+        expiredLockIds1.clear();
+        expiredLockIds2.clear();
+    }
+
+    private void clearLockTable(CorfuRuntime runtime) {
+        CorfuStore corfuStore = new CorfuStore(runtime);
+        Table<LockDataTypes.LockId, LockDataTypes.LockData, Message> lockTable =
+            corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, LOCK_TABLE_NAME);
+        lockTable.clearAll();
+    }
+
+    private void executeAcquireOrRenewConcurrently(LockStore lockStore1,
+        LockStore lockStore2, boolean renew) throws Exception {
+
+        if (renew) {
+            scheduleConcurrently(f -> {
+                try {
+                    renewed1 = lockStore1.renew(lockId);
+                } catch (LockStoreException le) {
+                    renewed1 = false;
+                }
+            });
+            scheduleConcurrently(f -> {
+                try {
+                    renewed2 = lockStore2.renew(lockId);
+                } catch (LockStoreException le) {
+                    renewed2 = false;
+                }
+            });
+        } else {
+            scheduleConcurrently(f -> {
+                try {
+                    acquired1 = lockStore1.acquire(lockId);
+                } catch (LockStoreException le) {
+                    acquired1 = false;
+                }
+            });
+
+            scheduleConcurrently(f -> {
+                try {
+                    acquired2 = lockStore2.acquire(lockId);
+                } catch (LockStoreException le) {
+                    acquired2 = false;
+                }
+            });
+        }
+        executeScheduled(2, TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    // Two runtimes try to acquire a lock with the same lock id concurrently.
+    // Verify that only one of them succeeds.
+    // Repeat this operation multiple times to expose race in this workflow,
+    // if any.  Each iteration clears the lock table and creates a new
+    // LockStore instance to exercise the init scenario.
+    @Test
+    public void testAcquireConcurrently() throws Exception {
+        for (int i=0; i<NUM_ITERATIONS; i++) {
+
+            LockStore lockStore1 = new LockStore(runtime1, UUID.randomUUID());
```

Comment:
I know we discussed this some time back, but can you remind me why is it that clearing the lock table is not enough?